### DISCUSSION TITLE
Checkstyle rule to verify "Requires" and "Provides"

### DISCUSSION THREAD

**Discussion by ndr-brt** [eclipse-edc/Connector#596] [Ideas]
I'm not expert in checkstyle, someone knows if it will be possible to write a check that verifies:
* for every `context.getService` invocation there should be a `@Requires` entry or a `@Inject field`
* for every `context.registerService` there should be a `@Provides` entry

Because it's really easy to forgot those annotations, this could be a way to avoid problems before merging code in the main branch

**Comment by paullatzelsperger** [comment]
there is a check directly in code which issues a warning, if the number of @Provides and `context.registerService` don't match, but there is none for the `@Requires` part. 

The big challenge with custom Checkstyle rules is that they practically cannot reside in the same repo. Or more specifically, they need to be on the classpath of checkstyle and accessed with their classname (AFAIK), which means they must be built _before_ the EDC gets built. That feels sort of clunky.

On way of going about this would be to create another repo in the `eclipse-dataspaceconnector` project and access the check as an artifact from there.

**Reply by ndr-brt** [reply]
Yes, seems pretty tricky, maybe we should just stick on with the warning.

**Reply by paullatzelsperger** [reply]
We may have an option with codeQL. Stay tuned, further info will come :)
